akka-actor-typed/src/main/scala/akka/actor/typed/BehaviorInterceptor.scala

```diff
@@ -17,6 +17,17 @@ import akka.annotation.{ DoNotInherit, InternalApi }
 abstract class BehaviorInterceptor[O, I] {
   import BehaviorInterceptor._
 
+  /**
+   * Allows for applying the interceptor only to certain message types. Useful if the official protocol and the actual
+   * protocol of an actor causes problems, for example class cast exceptions for a message not of type `O` that
+   * the actor still knows how to deal with. Note that this is only possible to use when `O` and `I` are the same type.
```

Comment:
Should we create a new subtype for this that enforces rather than null turning this off?

Reply:
I don't think that's needed, it's a power user feature that will likely not be used much, if at all, and it is only called by our internals, and _when_ it is used it is by the user overriding and not returning null, so not much risk for surprise-NPE:ing-users.